packages/extension/src/types/relay.ts

```diff
@@ -0,0 +1,47 @@
+import type { Infer } from 'superstruct';
+import { any, array, number, object, optional, pattern, string, union, unknown } from 'superstruct';
+
+export enum RelayTarget {
+  Page = 'Page',
+  Content = 'Content',
+  Background = 'Background',
+  External = 'External'
```

Comment:
What's this needed for?

Reply:
Two things:
- Depending on how messages are sent across browser extensions, they may not be fully isolated from other extensions. This basically checks if messages sent are coming from a part of the extension.
- This ensures that messages aren't replayed twice. Calling `window.postMessage` in the content script will also trigger the event listener in the content script.

Reply:
Sorry, I specifically meant `External`. I couldn't find it used anywhere?